src/Tribe/Aggregator/Processes/Import_Events.php

```diff
@@ -0,0 +1,365 @@
+	 *
+	 * @since TBD
+	 *
+	 * @return int
+	 */
+	public function get_record_id() {
+		return $this->record_id;
+	}
+
+	/**
+	 * Sets this import process record ID.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $record_id
+	 */
+	public function set_record_id( $record_id ) {
+		$this->record_id = $record_id;
+	}
+
+	/**
+	 * Handles the real import.
+	 *
+	 * In short: if an event has dependencies and those are not yet all in place then the event
+	 * will be re-queued; otherwise it's inserted.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $item
+	 *
+	 * @return array|false Either the event data to requeue or `false` if done.
+	 */
+	protected function task( $item ) {
+		$record_id             = $this->record_id = $item['record_id'];
+		$data                  = (array) $item['data'];
+		$this->transitional_id = filter_var( $item['transitional_id'], FILTER_SANITIZE_STRING );
+
+		$dependencies = $this->parse_linked_post_dependencies( $data );
+
+		if ( empty( $dependencies ) ) {
+			$this->has_dependencies = false;
+			$activity               = $this->insert_event( $record_id, (object) $data );
+			$this->activities[]     = $activity;
+
+			return $this->doing_sync ? $activity : false;
+		}
+
+		$dependencies_ids = $this->check_dependencies( $dependencies );
+
+		if ( $dependencies_ids ) {
+			$this->set_linked_posts_ids( $data, $dependencies_ids );
+			$activity           = $this->insert_event( $record_id, (object) $data );
+			$this->activities[] = $activity;
+
+			return $this->doing_sync ? $activity : false;
+		}
+
+		return $item;
+	}
+
+	/**
+	 * Parses the Event Venue and Organizer dependencies.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $data
+	 *
+	 * @return array An array containing a list of identifiers (contextual to the import) for the
+	 *               dependencies.
+	 */
+	protected function parse_linked_post_dependencies( $data ) {
+		$dependencies = array();
+
+		if ( ! empty( $data['depends_on'] ) ) {
+			$dependencies = array_values( (array) $data['depends_on'] );
+		}
+
+		return $dependencies;
+	}
+
+	/**
+	 * Inserts an event.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $record_id
+	 * @param array $data
+	 *
+	 * @return Tribe__Events__Aggregator__Record__Activity
+	 */
+	protected function insert_event( $record_id, $data ) {
+		$record = $this->get_record( $record_id );
+
+		if ( empty( $record ) || $record instanceof WP_Error ) {
+			// no point in going on
+			return false;
+		}
+
+		if ( ! $this->has_dependencies ) {
+			add_action( 'tribe_aggregator_after_insert_post', array( $this, 'add_transitional_data' ) );
+		}
+
+		$activity = $record->insert_posts( array( $data ) );
+		$record->activity()->merge( $activity );
+		$record->update_meta( 'activity', $record->activity() );
+
+		return $activity;
+	}
+
+	/**
+	 * Returns this import process record.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $record_id
+	 *
+	 * @return null|Tribe__Error|Tribe__Events__Aggregator__Record__Abstract
```

Comment:
I don't think it can return Tribe__Error (I feel like this mistake has kind of spread virally across the plugin code) ... Tribe__Error is used to generate WP_Error objects but generally instances of Tribe__Error aren't passed around.